dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/imports/programrule/ProgramRuleAssignActionTest.java

```diff
@@ -119,9 +130,62 @@ void shouldNotImportWithWarningWhenAttributeWithSameValueIsAssignedByAssignRule(
     assertHasOnlyWarnings(importReport, E1310);
   }
 
+  @Test
+  void shouldImportEventWithLastDateAndCorrectlyAssignPreviousEventDataValue() throws IOException {
```

Comment:
It's hard to see from this test what we are testing and what state is changing in the process 😅 

Why is it that you expect THIRD here and SECOND in the next test? I would like to see the difference in input/config that causes that inside the test.

Reply:
Done